examples/paint.rs

```diff
@@ -0,0 +1,275 @@
+    pub struct State {
+        is_dragging: bool,
+        previous_mouse_pos: Option<Point>,
+    }
+
+    impl<'a, Message> Widget<Message, Renderer> for Paint<'a, Message> {
+        fn width(&self) -> Length {
+            Length::Fill
+        }
+
+        fn height(&self) -> Length {
+            Length::Fill
+        }
+
+        fn layout(
+            &self,
+            _renderer: &Renderer,
+            limits: &layout::Limits,
+        ) -> layout::Node {
+            let size = limits
+                .height(Length::Fill)
+                .width(Length::Fill)
+                .resolve(Size::ZERO);
+            layout::Node::new(size)
+        }
+
+        fn draw(
+            &self,
+            _renderer: &mut Renderer,
+            layout: Layout<'_>,
+            _cursor_position: Point,
+        ) -> (Primitive, MouseCursor) {
+            let mut buffer: VertexBuffers<Vertex2D, u16> = VertexBuffers::new();
+            let mut path_builder = lyon::path::Path::builder();
+
+            let bounds = layout.bounds();
+
+            // Draw rectangle border with lyon.
+            basic_shapes::stroke_rectangle(
+                &lyon::math::Rect::new(
+                    lyon::math::Point::new(bounds.x + 0.5, bounds.y + 0.5),
+                    lyon::math::Size::new(
+                        bounds.width - 1.0,
+                        bounds.height - 1.0,
+                    ),
+                ),
+                &StrokeOptions::default().with_line_width(1.0),
+                &mut BuffersBuilder::new(
+                    &mut buffer,
+                    |pos: lyon::math::Point, _: StrokeAttributes| Vertex2D {
+                        position: pos.to_array(),
+                        color: [0.0, 0.0, 0.0, 1.0],
+                    },
+                ),
+            )
+            .unwrap();
+
+            for (from, to) in self.strokes {
+                path_builder.move_to(lyon::math::Point::new(
+                    from.x + bounds.x,
+                    from.y + bounds.y,
+                ));
+                path_builder.line_to(lyon::math::Point::new(
+                    to.x + bounds.x,
+                    to.y + bounds.y,
+                ));
+            }
+
+            let mut tessellator = StrokeTessellator::new();
+
+            // Draw strokes with lyon.
+            tessellator
+                .tessellate(
+                    &path_builder.build(),
+                    &StrokeOptions::default().with_line_width(3.0),
+                    &mut BuffersBuilder::new(
+                        &mut buffer,
+                        |pos: lyon::math::Point, _: StrokeAttributes| {
+                            Vertex2D {
+                                position: pos.to_array(),
+                                color: [0.0, 0.0, 0.0, 1.0],
+                            }
+                        },
+                    ),
+                )
+                .unwrap();
```

Comment:
The widget generates strokes every time the cursor is moved while dragging. It emulates a pencil. This means that strokes will add up very quickly and will be very close to each other.

I am not sure implementing a pencil like this is the best way to showcase how to combine the new `Mesh2D` primitive with `lyon`. My intution tells me that something like a buffer of pixels would work better for this particular case.

Maybe we could implement line/rectangle/circle tools instead?

Reply:
I agree with that "pencil" is not a very good example which demonstrates the combination of `iced` and `lyon`.
I choose a bezier tool as an example to demonstarate it.